components/switch/xiaomi.py

```diff
@@ -78,38 +55,20 @@ def turn_on(self, **kwargs):
         """Turn the switch on."""
         if self.xiaomi_hub.write_to_hub(self._sid, self._data_key, 'on'):
             self._state = True
-            self.schedule_update_ha_state()
 
     def turn_off(self):
         """Turn the switch off."""
         if self.xiaomi_hub.write_to_hub(self._sid, self._data_key, 'off'):
             self._state = False
-            self.schedule_update_ha_state()
-
-    def toggle(self):

```

Comment:
Already supported by ToggleEntity that we inherit. 
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L357